pkg/dependency/parser/dart/pub/parse.go

```diff
@@ -23,30 +24,50 @@ func NewParser() types.Parser {
 }
 
 type lock struct {
-	Packages map[string]Dep `yaml:"packages"`
+	Packages map[string]Dep    `yaml:"packages"`
+	Sdks     map[string]string `yaml:"sdks"`
 }
 
 type Dep struct {
-	Dependency string `yaml:"dependency"`
-	Version    string `yaml:"version"`
+	Dependency  string      `yaml:"dependency"`
+	Version     string      `yaml:"version"`
+	Source      string      `yaml:"source"`
+	Description Description `yaml:"description"`
 }
 
+type Description string
+
 func (Parser) Parse(r xio.ReadSeekerAt) ([]types.Library, []types.Dependency, error) {
 	l := &lock{}
 	if err := yaml.NewDecoder(r).Decode(&l); err != nil {
 		return nil, nil, xerrors.Errorf("failed to decode pubspec.lock: %w", err)
 	}
 	var libs []types.Library
 	for name, dep := range l.Packages {
+		// Some dependencies use one of the SDK versions.
+		// In this case dep.Version == `0.0.0`.
+		// We can't get versions for these dependencies.
+		// Therefore, we use the first version of the SDK constraint specified in the Description.
+		// See https://github.com/aquasecurity/trivy/issues/6017
+		version := dep.Version
+		if version == "0.0.0" && dep.Source == "sdk" {
+			if constraint, ok := l.Sdks[string(dep.Description)]; ok {
+				if v := firstVersionOfConstrain(constraint); v != "" {
+					log.Logger.Infof("The first version of %q constraint was used for %q.", dep.Description, name)
+					version = v
+				}
+			}
+		}
+
 		// We would like to exclude dev dependencies, but we cannot identify
 		// which indirect dependencies were introduced by dev dependencies
 		// as there are 3 dependency types, "direct main", "direct dev" and "transitive".
 		// It will be confusing if we exclude direct dev dependencies and include transitive dev dependencies.
 		// We decided to keep all dev dependencies until Pub will add support for "transitive main" and "transitive dev".
 		lib := types.Library{
-			ID:       pkgID(name, dep.Version),
+			ID:       utils.PackageID(name, dep.Version),
```

Comment:
```suggestion
			ID:       utils.PackageID(name, version),
```

Reply:
Thanks that you found my typo!
Fixed in https://github.com/aquasecurity/trivy/pull/6239/commits/c5ce4bc138969265cca1fd58f00664fa645e827d